doc/sfp-cmis/cmis.md

```diff
@@ -0,0 +1,308 @@
+
+# Requirement Overview
+
+This document describes functional behavior of the QSFPDD CMIS support in SONiC.
+
+The QSFPDD Common Management Interface Specification (CMIS) version 4.0 was finalized
+in May of 2019 and provides a variety of features and support for different transceiver
+form factors. From a software perspective, a CMIS application initialization sequence
+is now mandatory as per the current dynamic port breakout mode activated. And these is
+also a new diagnostic support tha could be useful to SONiC users and developers.
+
+The current SONiC platform API model is such that base classes in **sonic-platform-common**
+define a common, platform-independent interface that can be implemented in platform-specific classes.
+
+![](images/001.png)
+
+Unfortunately, the legacy SFP libraries (sonic-sfp) do not provide full featured support for the SFP,
+and and vendors are having to fill in implementation gaps for themselves. This is resulting in a lot
+of duplicate work being done for things that should be common across platforms.
+
+The goal of this SONIC QSFPDD CMIS support is to
+
+- Provide an unified common SFP parser for the QSFPDD transceivers
+- Enhance the **pmon#xcvrd** for the QSFPDD application initialization sequence
+- Enhance the **pmon#xcvrd** for the QSFPDD diagnostics loopback controls
+
+## 1.1 Functional Requirements
+
+1. Ability to parse the advertised applications from the transceivers
+2. Ability to post the advertised applications to the STATE_DB
+3. Ability to decode the current module state from the transceivers
+4. Ability to post the current module state to the STATE_DB
+5. Ability to decode the error code of the application initialization sequence from the transceivers
+6. Ability to post the error code of the application initialization sequence
+7. Ability to process the application initialization sequence in multi-thread/process model.
+8. Ability to enable / disable the transceiver loopback controls
+
+**Note:**  
+The duration of the CMIS application initialization sequence greatly differs from transceivers
+to transceivers, while some take 3 seconds for activating the 4x100G mode, some require 15 seconds.
+Hence it's mandatory to support multi-threaded/processed models.
+
+## 1.2 Configuration and Management Requirements
+
+1. Show commands to display the QSFPDD EEPROM and DOM information, as described in **Show Commands** section below.
+
+## 1.3 Warm Boot Requirements
+
+Functionality should continue to work across warm boot.
+- To support planned system warm boot.
+- To support pmon docker restart.
+
+# Functionality
+
+## 2.1 Target Deployment Use Cases
+The QSFPDD CMIS support is used to perform application initialization as per the operational
```

Comment:
Any case where the configured module speed is not the same as the default advertised application would require this.

Reply:
Yes, exactly, and thanks for the comments, the HLD will be updated with more details